src/realm/query_engine.hpp

```diff
@@ -1972,6 +1972,48 @@ class StringNode<EqualIns> : public StringNodeEqualBase {
     size_t _find_first_local(size_t start, size_t end) override;
 };
 
+
+class LinkMap;
+class StringNodeFulltext : public StringNodeEqualBase {
+public:
+    StringNodeFulltext(StringData v, ColKey column, std::unique_ptr<LinkMap> lm = {});
+
+    void table_changed() override;
+
+    void _search_index_init() override;
+
+    std::unique_ptr<ParentNode> clone() const override
+    {
+        return std::unique_ptr<ParentNode>(new StringNodeFulltext(*this));
+    }
+
+    virtual std::string describe_condition() const override
+    {
+        return "FULLTEXT";
+    }
+
+    const std::vector<ObjKey>& index_based_keys() override
```

Comment:
can be `const`

Reply:
I think there is one function that actually modifies the state.